### PR TITLE
[WIP][monorepo] Typings refactor

### PR DESCRIPTION
**Pull request by cf19drofxots**
### Description

This refactors the typings used across the packages to remove weird placements and dependencies currently placed on the machine package.

There's not much going on here logically. The main thing this PR does is it moves most of the types from the `machine` to separate namespaces inside of `cf.js`

### CODE REVIEW

**Comment by cf19drofxots** [comment]
@snario I'm currently not touching the semantics of the refactoring whatsoever. Just moving things around to ensure consumption compatibility is sane.

This PR is also gonna get even bigger than it currently is, and I don't want to tackle everything related to this issue in one PR (as merging master / rebasing) becomes an even bigger pain.

This is the first of a series of PRs (as originally intended to keep PRs as small as possible).

So it's probably best to keep most of the semantic comments for a follow up PR.

**Comment by snario** [comment]
> @snario I'm currently not touching the semantics of the refactoring whatsoever. Just moving things around to ensure consumption compatibility is sane.
> 
> This PR is also gonna get even bigger than it currently is, and I don't want to tackle everything related to this issue in one PR (as merging master / rebasing) becomes an even bigger pain.
> 
> This is the first of a series of PRs (as originally intended to keep PRs as small as possible).
> 
> So it's probably best to keep most of the semantic comments for a follow up PR.

SGTM, I'd recommend noting down the comments somewhere though, maybe making issues

**Comment by cf19drofxots** [comment]
Issues seem appropriate

**Comment by snario** [comment]
[ch22]

File: packages/cf.js/src/app/index.ts [hunk 1]
```diff
@@ -13,6 +13,8 @@ import {
 import * as abi from "../utils/abi";
 import { CfNonce } from "../utils/nonce";
 
+const { keccak256 } = ethers.utils;
```
Comment (snario) [inline packages/cf.js/src/app/index.ts]: If you're doing this let's just get rid of the `keccak256` export in `abi.ts`

Comment (cf19drofxots) [reply packages/cf.js/src/app/index.ts]: I did

File: packages/cf.js/src/app/index.ts [hunk 2]
```diff
@@ -31,6 +33,7 @@ export class CfAppInstance {
   ) {}
 
   public cfAddress(): H256 {
+    // FIXME: shouldn't have to require abi and bytecode here
```
Comment (snario) [inline packages/cf.js/src/app/index.ts]: This is a large FIXME and is encompassed with the general direction of the project and is not necessarily just a purely technical fix (requires architectural changes too). For this reason I vote that we remove the FIXMEl

File: packages/cf.js/src/utils/abi.ts [hunk 3]
```diff
@@ -11,7 +11,3 @@ export function decode(types: string[], data: ethers.utils.Arrayish) {
 export function encodePacked(types: string[], values: any[]) {
   return ethers.utils.solidityPack(types, values);
 }
-
-export function keccak256(encodePacked: string): string {
-  return ethers.utils.keccak256(encodePacked);
-}
```
Comment (snario) [inline packages/cf.js/src/utils/abi.ts]: Oh you did. Cool.

**Pull request merged**